### PR TITLE
added fuzzy searching by name

### LINKED ISSUE #10: Fuzzy search and satelitte names
After #9 is done, add ability for user to search satelitties by satelitte name with fuzzy search

### PR DESCRIPTION
**Pull request by jpitucha**
resolves #10

### CODE REVIEW

File: src/utils/satelliteUtils.ts [hunk 1]
```diff
@@ -3,16 +3,20 @@ import SatelliteInterface from '../utils/types/satelliteInterface'
...
     }
 
     static getSatelliteById(id: string): Promise<SatelliteDocument | null> {
         return Satellite.findById(id).exec();
     }
 
-    static getSatelliteByName(name: string): Promise<SatelliteDocument | null> {
-        return Satellite.findOne({ officialName: name }).exec()
+    static async getSatelliteByName(name: string): Promise<SatelliteDocument | null> {
+        const foundSatellites = await Satellite.fuzzySearch(name);
```
Comment (michaelKurowski) [inline src/utils/satelliteUtils.ts]: Would be probably better to return a list of satellities.

**Review by michaelKurowski — APPROVED** [review]

File: src/utils/satelliteUtils.ts (hunk 1 above)
Comment (michaelKurowski) [reply src/utils/satelliteUtils.ts]: Fix it, I'm approving in advance as the code seems fine.

File: src/vendor.d.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    interface MongooseFuzzySearchingMiddlewares<T = unknown> {
+        preSave(this: Query<T>): void;
+        preInsertMany(this: Query<T>): Promise<void>;
+        preUpdate(this: Query<T>): Promise<void>;
+        preUpdateOne(): Promise<void>;
+        preFindOneAndUpdate(this: Query<T>): Promise<void>;
+        preUpdateMany(this: Query<T>): Promise<void>;
+    }
+
+    interface MongooseFuzzySearchingOptions<T extends Record<string, unknown>> {
```
Comment (michaelKurowski) [inline src/vendor.d.ts:22]: Very good that you used `unknown`. Good job

File: src/utils/satelliteUtils.ts (hunk 1 above)
Comment (Jordan141) [reply src/utils/satelliteUtils.ts]: +1 here but I'm not approving until it's done

**Review by Jordan141 — CHANGES_REQUESTED** [review]

**Review by Jordan141 — APPROVED** [review]

File: src/routes/local.ts [hunk 3]
```diff
@@ -26,7 +27,9 @@ router.get('/satellite', (req, res) => {
         return SatelliteUtilities.getSatelliteByName(req.query.name)
             .then((satelliteDoc) => {
                 if (!satelliteDoc) return res.sendStatus(400)
-                return res.json(_.pick(satelliteDoc, PROPS_TO_SEND_IN_RESPOSNE))
+                const satellitesArray = <Partial<SatelliteInterface>[]> []
```
Comment (ixcviw7bw) [inline src/routes/local.ts]: You can use Pick<SatelliteInterface, "officialName" | "_id" ...> instead of Partial<>

File: src/routes/local.ts [hunk 4]
```diff
@@ -26,7 +27,9 @@ router.get('/satellite', (req, res) => {
...
+                satelliteDoc.map((satellite) => satellitesArray.push(_.pick(satellite, PROPS_TO_SEND_IN_RESPOSNE)))
```
Comment (ixcviw7bw) [inline src/routes/local.ts]: Either change .map to .forEach, or remove .push from inside .map and assign .map output to satellitesArray.

**Review by ixcviw7bw — APPROVED** [review]

**Pull request merged**